app/models/event.rb

```diff
@@ -504,6 +504,39 @@ def self.subj_id_check(options = {})
     end
   end
 
+  def self.modify_nested_objects(options = {})
+    size = (options[:size] || 1000).to_i
+    cursor = [options[:from_id], options[:until_id]]
+
+    response = Event.query(nil,  page: { size: 1, cursor: [] })
+    Rails.logger.warn "[modify_nested_objects] #{response.results.total} events for source datacite-crossref."
+
+    # walk through results using cursor
+    if response.results.total.positive?
+      while response.results.results.length.positive?
+        response = Event.query(nil, page: { size: size, cursor: cursor })
+        break unless response.results.results.length.positive?
+
+        Rails.logger.warn "[modify_nested_objects] modify_nested_objects #{response.results.results.length}  events starting with _id #{response.results.to_a.first[:_id]}."
+        cursor = response.results.to_a.last[:sort]
+        Rails.logger.warn "[modify_nested_objects] Cursor: #{cursor} "
+
+        ids = response.results.results.map(&:uuid).uniq
+        CamelcaseNestedObjectsJob.perform_later(ids, options)
+      end
+    end
+  end
+
+
+  def self.camelcase_nested_objects(uuid)
+      event = Event.find_by(uuid: uuid)
+      if event.present?
+        subj = event.subj.transform_keys { |key| key.to_s.underscore.camelcase(:lower) } 
```

Comment:
How does this work for nested keys in `subj` and `obj`, or do we not care?

Reply:
i am not sure if I follow you question; this method transforms  the nested keys in `subj` and `obj` to `camelcase`.  

I have added a test in case that clarifies further how it works

Reply:
@mfenner this is the only pending question. any further comments?

Reply:
Thanks. Which test clarifies how it works?

Reply:
https://github.com/datacite/lupo/pull/454/commits/b6679341975c23c4c53c458442040f4996bf5e3f

Reply:
and https://github.com/datacite/lupo/pull/454/commits/7d6102cdfb7876b9645140c3ec9ee4f1fa67c076

Reply:
Thanks. Then I think we use different language, as `subj` and `obj` in the tests are not nested objects, but objects without nesting. My whole point was that `transform_keys` does not work on nested objects.

Please go ahead and merge the pull request.